packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -338,24 +399,70 @@ function WorklistDataGrid(props) {
     selectCell(cellRowIdColId, cellCoordinates);
   };
 
-  const handleRowSelection = (rowId, rowIndex, selectedCellCoordinates) => {
+  const handleRowSelection = (rowId, rowIndex, selectedCellCoordinates, enableSelectableRows, useLastSelectedRowAsAnchor) => {
     handleCellSelectionChange(null, null, selectedCellCoordinates);
-    selectRows(false, rowId, rowIndex);
+    const selectionData = useLastSelectedRowAsAnchor ? multiRowSelectionWithLastSelected?.current : multiRowSelection?.current;
+
+    // TODO: Clean up the if condition.
+    if (!enableSelectableRows) {
+      if (hasSelectableRows) {
+        // regular click or space key
+        selectRows(false, rowId, rowIndex);
+      }
+    } else if (hasSelectableRows) {
+      // Shift key is pressed
+      if (selectionData?.anchorRow) {
+        // ACTION: select multiple rows based on anchor
+        handleEnableSelectableRows(rowIndex, selectionData);
+        selectionData.previousSelectionEndRow = rowIndex;
+      } else {
+        // no anchor row
+        selectRows(false, rowId, rowIndex);
+      }
+    } else if (useLastSelectedRowAsAnchor) {
+      // Shift + Space
+      isRowSelectionEnabledByGrid.current = !hasSelectableRows;
+      multiRowSelectionWithLastSelected.current = { anchorRow: rowIndex };
+      handleEnableSelectableRows(rowIndex, multiRowSelectionWithLastSelected.current);
+      multiRowSelectionWithLastSelected.current.previousSelectionEndRow = rowIndex;
+    }
+  };
+
+  const handleKeyUp = (event) => {
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_SHIFT:
+        multiRowSelection.current = null;
+        break;
+      default:
+        break;
+    }
   };
 
   const handleKeyDown = (event) => {
     const cellCoordinates = { row: focusedRow.current, col: focusedCol.current };
     let nextRow = cellCoordinates.row;
     let nextCol = cellCoordinates.col;
-
+    let activateMultiRowSelection = false;
     const key = event.keyCode;
     switch (key) {
       case KeyCode.KEY_UP:
+        if (event.shiftKey) {
+          activateMultiRowSelection = true;
+        }
         nextRow -= 1;
         break;
       case KeyCode.KEY_DOWN:
+        if (event.shiftKey) {
+          activateMultiRowSelection = true;
+        }
         nextRow += 1;
         break;
+      case KeyCode.KEY_SHIFT:
```

Comment:
The keydown event has a Boolean check for if the shift key is pressed.  You should not need a handler for the shift key.